flatpak/manifest_tool/sources_ops.py

```
@@ -0,0 +1,409 @@
+                width=120
+            )
+            path.write_text(new_content, encoding="utf-8")
+            _LOGGER.debug("Replaced url with path for %s", identifier)
+
+        return changed
+
+    except (FileNotFoundError, OSError) as e:
+        _LOGGER.warning("Failed to process manifest %s: %s", manifest_path, e)
+        return None
+    except yaml.YAMLError as e:
+        _LOGGER.error("Failed to parse YAML in %s: %s", manifest_path, e)
```

Ruff (0.13.1)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)